src/test/java/bio/terra/service/dataset/DatasetBucketDaoTest.java

```diff
@@ -0,0 +1,246 @@
+import bio.terra.common.Table;
+import bio.terra.common.category.Unit;
+import bio.terra.common.fixtures.JsonLoader;
+import bio.terra.common.fixtures.ProfileFixtures;
+import bio.terra.common.fixtures.ResourceFixtures;
+import bio.terra.model.BillingProfileModel;
+import bio.terra.model.BillingProfileRequestModel;
+import bio.terra.model.DatasetRequestModel;
+import bio.terra.service.dataset.exception.DatasetLockException;
+import bio.terra.service.dataset.exception.DatasetNotFoundException;
+import bio.terra.service.profile.ProfileDao;
+import bio.terra.service.resourcemanagement.ResourceService;
+import bio.terra.service.resourcemanagement.google.GoogleBucketResource;
+import bio.terra.service.resourcemanagement.google.GoogleProjectResource;
+import bio.terra.service.resourcemanagement.google.GoogleResourceDao;
+import org.hamcrest.Matchers;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest
+@AutoConfigureMockMvc
+@Category(Unit.class)
+public class DatasetBucketDaoTest {
+
+    private static final Logger logger = LoggerFactory.getLogger(DatasetBucketDaoTest.class);
+
+    @Autowired
+    private JsonLoader jsonLoader;
+
+    @Autowired
+    private DatasetDao datasetDao;
+
+    @Autowired
+    private DatasetBucketDao datasetBucketDao;
+
+    @Autowired
+    private ProfileDao profileDao;
+
+    @Autowired
+    private GoogleResourceDao resourceDao;
+
+    @Autowired
+    private NamedParameterJdbcTemplate jdbcTemplate;
+
+    @Autowired
+    private ResourceService resourceService;
+
+    private BillingProfileModel billingProfile;
+    private GoogleBucketResource bucketForFile;
+    private UUID projectId;
+    private Dataset dataset;
+
+    private UUID datasetId;
+    private UUID bucketResourceId;
+
+    private UUID createDataset(DatasetRequestModel datasetRequest, String newName) throws Exception {
```

Comment:
Typically, private helper methods are at the bottom of the class

Reply:
Moved!